src/explore-education-statistics-common/src/modules/table-tool/contexts/TableHeadersContext.tsx

```diff
@@ -0,0 +1,109 @@
+import useToggle from '@common/hooks/useToggle';
+import noop from 'lodash/noop';
+import React, {
+  createContext,
+  ReactNode,
+  useContext,
+  useMemo,
+  useState,
+} from 'react';
+
+export interface TableHeadersContextState {
+  activeList?: string;
+  defaultNumberOfItems: number;
+  expandedLists: string[];
+  groupDraggingActive: boolean;
+  groupDraggingEnabled: boolean;
+  setActiveList: (listName?: string) => void;
+  toggleExpandedList: (listName: string) => void;
+  toggleGroupDraggingActive: () => void;
```

Comment:
Would suggest that this requires an explicit boolean parameter like `active`.

The main problem with just calling the toggle function without one is that if the user does things too quickly, the toggle state can end up inverted and the UI would end up looking messed up.